### ISSUE TITLE
[bug] Targets with failed `deps` are still run during `moon ci`

### ISSUE THREAD

**Issue by ilyasotkov** [moonrepo/moon#926]
**Describe the bug**
Targets with failed `deps` are triggered with `moon ci` but not `moon run`

**Steps to reproduce**

With this `moon.yml`:
```yaml
tasks:
  build:
    command: exit 1
    options:
      shell: true
      cache: false
  deploy:
    deps:
    - build
    command: echo 'Deployed!'
    options:
      shell: true
      cache: false
```
If I run `moon ci`
```diff
▪▪▪▪ Gathering touched files
  apps/myapp/moon.yml
  apps/myapp/out.txt
▪▪▪▪ Gathering runnable targets
  myapp:build
  myapp:deploy
▪▪▪▪ Generating dependency graph
Target count: 2
Action count: 6
▪▪▪▪ Running all targets
▪▪▪▪ myapp:build
▪▪▪▪ myapp:build (5ms)
▪▪▪▪ myapp:deploy
myapp:deploy | Deployed!
▪▪▪▪ myapp:deploy (5ms)
▪▪▪▪ Results

pass SetupSystemTool (skipped)
pass InstallSystemDeps (skipped)
pass SyncSystemProject(myapp) (0s)
fail RunTarget(myapp:build) (6ms)
     Process exit failed with a 1 exit code.
pass SyncWorkspace (0s)
pass RunTarget(myapp:deploy) (6ms)


Actions: 5 completed, 1 failed
   Time: 13ms
```

Note this

```
▪▪▪▪ myapp:deploy
myapp:deploy | Deployed!
▪▪▪▪ myapp:deploy (5ms)
```

**Expected behavior**

If I run `moon run :build :deploy` or `moon run :deploy`, myapp:deploy is never run because its dependency, myapp:build, failed:

```txt
▪▪▪▪ myapp:build
▪▪▪▪ myapp:build (4ms)
Error:   × Process exit failed with a 1 exit code.
```

**Environment**
moon 1.8.1

**Comment by milesj** [comment]
@ilyasotkov This is kind of by design right now, but I agree it's not ideal and I need to rebuild it.

`moon run` works because we abort on the 1st failure, so subsequent targets are not run. `moon ci` runs everything and never aborts, so that all failing targets can be aggregated and reviewed.

**Comment by milesj** [comment]
Fixed in v1.10!
